substrate/frame/metadata-hash-extension/src/lib.rs

```diff
@@ -68,12 +68,20 @@ enum MetadataHash {
 	Custom([u8; 32]),
 }
 
+const RUNTIME_METADATA: Option<[u8; 32]> = if let Some(hex) = option_env!("RUNTIME_METADATA_HASH") {
+	match const_hex::const_decode_to_array(hex.as_bytes()) {
+		Ok(hex) => Some(hex),
+		Err(_) => None,
```

Comment:
I cannot find a way to print some error in const fn, but this behavior is unchanged.

Reply:
You can panic in const? That should be good enough here?

Reply:
Yes, done in https://github.com/paritytech/polkadot-sdk/pull/5277/commits/8ffd8a91b995a78882b2935ad1c8ae87565137d3

Note that I can't put the exact error in the panic because it only accepts a string literal.